apps/studio/src/components/PageEditor/LinkEditorModal.tsx

```diff
@@ -125,48 +93,37 @@ const LinkEditorModalContent = ({
               type="text"
               placeholder="Browse grants"
               {...register("linkText")}
+              isInvalid={showLinkTextErrorState}
+              onChange={(e) => {
+                setIsFirstRender(false)
+                setValue("linkText", e.target.value, { shouldValidate: true })
+              }}
             />
 
-            {errors.linkText?.message && (
-              <FormErrorMessage>{errors.linkText.message}</FormErrorMessage>
+            {showLinkTextErrorState && (
+              <FormErrorMessage>{errors.linkText?.message}</FormErrorMessage>
             )}
           </FormControl>
 
           <Box mt="1.5rem">
             <LinkHrefEditor
               value={watch("linkHref")}
-              onChange={(value) => setValue("linkHref", value)}
+              onChange={({ value, shouldValidate }) =>
+                setValue("linkHref", value, { shouldValidate })
+              }
               label="Link destination"
               description="When this is clicked, open:"
               isRequired
               isInvalid={!!errors.linkHref}
-              pageLinkElement={
-                <PageLinkElement
-                  value={watch("linkHref")}
-                  onChange={(value) => setValue("linkHref", value)}
-                />
-              }
-              fileLinkElement={
-                <FileAttachment
-                  siteId={siteId}
-                  error={errors.linkHref?.message}
-                  setError={(errorMessage) =>
-                    setError("linkHref", {
-                      type: "custom",
-                      message: errorMessage,
-                    })
-                  }
-                  clearError={() => clearErrors("linkHref")}
-                  setHref={(linkHref) => {
-                    setValue("linkHref", linkHref)
-                  }}
-                />
+              errorMessage={errors.linkHref?.message}
+              setErrorMessage={(errorMessage) =>
+                setError("linkHref", {
+                  type: "custom",
+                  message: errorMessage,
+                })
               }
+              clearErrorMessage={() => clearErrors("linkHref")}
```

Comment:
sorry for this but i removed all these props from `FileAttachment` so might no longer be required here